### PR TITLE
Fix peeps leaving shop at upward slope.

### PR DESCRIPTION
**Pull request by tylerleamon**
Fix: https://github.com/OpenRCT2/OpenRCT2/issues/10634#event-3273204131

Fix follows logic for peeps exiting rides in UpdateRideLeaveExit().

Originally Peeps, once exiting the shop, would double-back towards the shop before resuming walking. In RCT Classic, Peeps still double-back, but find the sloped path. Not able to verify if issue exists in Vanilla where Peeps get stuck instead of finding path. Either way, this fix prevents Peeps from doubling-back towards the shop.

Included test file. Tests Toilets, First Aid, and Information Kiosk using both slope directions and flat. Use Cheats with Large Tram to fill park. Then execute Max Toilets, Nausea, or Rain. 
[Test_Shop.sv6.zip](https://github.com/OpenRCT2/OpenRCT2/files/4638456/Test_Shop.sv6.zip)

### CODE REVIEW

File: src/openrct2/peep/Guest.cpp [hunk 1]
```diff
@@ -5113,7 +5113,7 @@ void Guest::UpdateRideShopLeave()
             return;
     }
 
-    SetState(PEEP_STATE_WALKING);
+    SetState(PEEP_STATE_FALLING);
```
Comment (Gymnasiast) [inline src/openrct2/peep/Guest.cpp:5116]: Put a comment above this saying it used to be `PEEP_STATE_WALKING`, and why it was changed. Link to the issue as well.

Especially for the black magic that is pathfinding, I prefer if changes are properly documented. :-)

Comment (tylerleamon) [reply src/openrct2/peep/Guest.cpp:5117]: Updated. At this line in the pathfinding, Peeps are at the exit point of the shop/ride. It isn't a perfect solution to set them to falling, but it does prevent them from walking until they've truly exited the shop/ride.

**Comment by duncanspumpkin** [comment]
Yes that seems pretty reasonable. Have you checked a variety of situations and types of shop to ensure that they do not fall through the floor?

Will need a replay update thats why CI's are failing

**Comment by Gymnasiast** [comment]
We should also test this above water, preferably with normal path, an upwards slope and a downwards slope. Just to make sure that the peeps will not drown.

**Comment by tylerleamon** [comment]
> Yes that seems pretty reasonable. Have you checked a variety of situations and types of shop to ensure that they do not fall through the floor?
> 
> Will need a replay update thats why CI's are failing

I'll do some testing and upload a new save file that includes testing above water and slopes that do/don't have land underneath.

**Comment by tylerleamon** [comment]
Testing went well. I loaded 750 guests in the attached park with a variety of slopes and terrain with water, land, underground, above ground. No guests drowned or were lost to the abyss. I also stepped through the UpdateFalling() function when a peep exits a toilet on an upward slope directly over water. The potential for drowning exists until the Line 938 Condition:
`                    if (height - 4 >= z && height < z + 20)
`
Because the `height` of the water is equal to the Peep's `z`, the drowning condition fails and the loop attempts to find the next tile element at the Peep's location, if it exists, which is the sloped path. The loop finds the sloped path and the peep enters the walking state.

Also a quick note. UpdateFalling() has the comment `Note: peeps 'fall' after leaving a ride/enter the park.`. Looks like this logic has been in the code for a while for peeps entering the park.

[Test_Shop_Terrain.sv6.zip](https://github.com/OpenRCT2/OpenRCT2/files/4641149/Test_Shop_Terrain.sv6.zip)

**Comment by duncanspumpkin** [comment]
Yes I wrote that comment. I think most likely CS just thought there was no need to do a fall for shops as it could mean that peeps return to the shop again. I'll fix the replay when I get a moment. This will requrie a network increment as well.

**Comment by Gymnasiast** [comment]
I'd like to test it as well.

**Comment by Gymnasiast** [comment]
I have tested it. It seems to work correctly. :+1:

It will need a changelog entry.
`- Fix: [#10634] Guests are unable to use uphill paths out of toilets.`

**Pull request merged**